Unhandled exception in autosave cause AutoSaveJob to stop

**Describe the bug**
Unhandled exception in AutoSaveJob cause process to stop and saving no longer performed.
```
!ENTRY org.eclipse.core.jobs 4 2 2021-10-31 13:30:02.047
!MESSAGE An internal error occurred during: "Saving backup file".
!STACK 0
com.thoughtworks.xstream.io.StreamException:  : Insufficient system resources exist to complete the requested service
	at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:77)
	at com.thoughtworks.xstream.core.util.QuickWriter.write(QuickWriter.java:49)
	at com.thoughtworks.xstream.io.xml.PrettyPrintWriter.writeText(PrettyPrintWriter.java:279)
	at com.thoughtworks.xstream.io.xml.PrettyPrintWriter.writeAttributeValue(PrettyPrintWriter.java:227)
	at com.thoughtworks.xstream.io.xml.PrettyPrintWriter.addAttribute(PrettyPrintWriter.java:222)
	at com.thoughtworks.xstream.io.WriterWrapper.addAttribute(WriterWrapper.java:41)
	at name.abuchen.portfolio.util.XStreamSecurityPriceConverter.marshal(XStreamSecurityPriceConverter.java:27)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:69)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:43)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:88)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:64)
	at com.thoughtworks.xstream.converters.collections.CollectionConverter.marshal(CollectionConverter.java:74)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:69)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:84)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshallField(AbstractReflectionConverter.java:256)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.writeField(AbstractReflectionConverter.java:232)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.<init>(AbstractReflectionConverter.java:195)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doMarshal(AbstractReflectionConverter.java:141)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:89)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:69)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:43)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:88)
	at com.thoughtworks.xstream.converters.collections.AbstractCollectionConverter.writeItem(AbstractCollectionConverter.java:64)
	at com.thoughtworks.xstream.converters.collections.CollectionConverter.marshal(CollectionConverter.java:74)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:69)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller$1.convertAnother(AbstractReferenceMarshaller.java:84)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshallField(AbstractReflectionConverter.java:256)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.writeField(AbstractReflectionConverter.java:232)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.<init>(AbstractReflectionConverter.java:195)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doMarshal(AbstractReflectionConverter.java:141)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:89)
	at com.thoughtworks.xstream.core.AbstractReferenceMarshaller.convert(AbstractReferenceMarshaller.java:69)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:43)
	at com.thoughtworks.xstream.core.TreeMarshaller.start(TreeMarshaller.java:82)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.marshal(AbstractTreeMarshallingStrategy.java:37)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:1043)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:1032)
	at com.thoughtworks.xstream.XStream.toXML(XStream.java:1005)
	at name.abuchen.portfolio.model.ClientFactory$PlainWriter.save(ClientFactory.java:150)
	at name.abuchen.portfolio.model.ClientFactory.save(ClientFactory.java:453)
	at name.abuchen.portfolio.ui.editor.ClientInput.autoSave(ClientInput.java:300)
	at name.abuchen.portfolio.ui.jobs.AutoSaveJob.run(AutoSaveJob.java:30)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.io.IOException: Insufficient system resources exist to complete the requested service
	at java.base/java.io.FileOutputStream.writeBytes(Native Method)
	at java.base/java.io.FileOutputStream.write(Unknown Source)
	at java.base/java.io.BufferedOutputStream.flushBuffer(Unknown Source)
	at java.base/java.io.BufferedOutputStream.flush(Unknown Source)
	at java.base/sun.nio.cs.StreamEncoder.implFlush(Unknown Source)
	at java.base/sun.nio.cs.StreamEncoder.flush(Unknown Source)
	at java.base/java.io.OutputStreamWriter.flush(Unknown Source)
	at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:75)
	... 47 more
```

**To Reproduce**
This issue hard to reproduce and usually caused by external factors, like machine goes sleep, internet connection issues and etc. They might be temporal or semi permanent (stop when something else is fixed)
In my case it was caused by some issues with Google Drive connected device

**Expected behavior**
Desired behavior is job like autosave should continue attempts to save file even if getting unexpected errors.
[there](https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/jobs/AutoSaveJob.java#L30) should be try catch for all non fatal errors (Fatal error is such as *Error or InterruptedException) 

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 0.55.0

--- a/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/jobs/AutoSaveJob.java
+++ b/name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/jobs/AutoSaveJob.java
@@ -26,9 +26,15 @@ public final class AutoSaveJob extends AbstractClientJob
         // 0 means not to autosave at all
         if (delay != 0L)
         {
-            PortfolioPlugin.info("Auto-saving " + clientInput.getLabel()); //$NON-NLS-1$
-            this.clientInput.autoSave();
-            schedule(delay);
+            try
+            {
+                PortfolioPlugin.info("Auto-saving " + clientInput.getLabel()); //$NON-NLS-1$
+                this.clientInput.autoSave();
+            }
+            finally
+            {
+                schedule(delay);
+            }
         }
         return Status.OK_STATUS;
     }

